packages/generator-feathers/test/generators.using.jest.test.js

```diff
@@ -0,0 +1,125 @@
+    }
+  };
+
+  child.stdout.on('data', addToBuffer);
+  child.stderr.on('data', addToBuffer);
+
+  child.on('exit', status => {
+    if(status !== 0) {
+      return reject(new Error(buffer));
+    }
+    resolve({ buffer, child });
+  });
+});
+
+const delay = ms => res => new Promise(resolve => setTimeout(() => resolve(res), ms));
+
+describe('generator-feathers with jest', () => {
+
+  let appDir;
+
+  const runTest = expectedText => 
+    startAndWait('npm', ['test'], { cwd: appDir }).then(({ buffer }) => {
+      assert.ok(buffer.indexOf(expectedText) !== -1,
+        'Ran test with text: ' + expectedText);
+    });
+
+  beforeEach(() =>
+    helpers
+      .run(path.join(__dirname, '..', 'generators', 'app'))
+      .inTmpDir(dir => appDir = dir)
+      .withPrompts({
+        name: 'my_jest_app',
+        providers: ['rest', 'socketio'],
+        src: 'src',
+        packager: 'npm@>= 3.0.0',
+        tester: 'jest'
+      }) // choose 'jest'
+      .withOptions({ skipInstall: false })
+  );
+
+  it('feathers:app', () => 
+    runTest('Feathers application tests (with jest').then(() => {
+      const pkg = require(path.join(appDir, 'package.json'));
+      const cfg = require(path.join(appDir, '.eslintrc.json'));
+      assert.ok(pkg.devDependencies.jest, 'Added jest as a devDependency');
+      assert.ok(cfg.env.jest, 'ESLint configured to support jest');
+    })
+  );
+
+  it('feathers:hook', () => helpers
+    .run(path.join(__dirname, '../generators/hook'))
+    .inTmpDir(() => process.chdir(appDir))
+    .withPrompts({ name: 'captain', type: 'before', services: [] })
+    .then(() => runTest('test/hooks/captain.test.js'))
+  );
+
+  describe('feathers:service', () => {
+    const testServiceGenerator = (adapter, database, id = null) => {
+      return helpers
+      .run(path.join(__dirname, '../generators/service'))
+      .inTmpDir(() => process.chdir(appDir))
+      .withPrompts({ adapter, database, name: adapter, path: adapter })
+      .withOptions({ skipInstall: false })
+      .then(() => runTest(`test/services/${adapter}.test.js`))
+      .then(() => startAndWait('node', ['src/'], { cwd: appDir }, 'Feathers application started'))
+      .then(delay(3000))
+      .then(({ child }) => {
+        const text = 'This is a jest.';
+        const body = { text };
+        if ('cassandra' === database) {
+          body.id = 1;
+        }
+        return rp({
+          url: `http://localhost:3030/${adapter}`,
+          method: 'post',
+          json: true,
+          body
+        })
+        .then(res => {
+          if (id) {
+            assert.ok(typeof res[id] !== 'undefined');
+          }
+          assert.equal(res.text, text);
+        })
+        .then(() => child.kill())
+        .catch(e => new Promise((resolve, reject) => {
+          child.once('exit', () => reject(e));
+          child.kill('SIGKILL')
+        }));
+      });
+    };
+
+    it('generic',   () => testServiceGenerator('generic'));
```

Comment:
We can probably remove the other database tests here and assume everything will work fine to not make the tests run a little quicker.

Reply:
Cool. Skipping all but `generic` test reduced overall testing times on travis from over 40 minutes down to about 15 minutes. Not sure why the latest build failed, but the offending code was the `authentication` package, not the generators. Also, you may already know this, but apparently you can tell `lerna` to [only run the tests for the packages that were most recently updated](https://github.com/lerna/lerna/issues/1263). ([official docs](https://github.com/lerna/lerna/tree/master/core/filter-options)) I had never heard of `lerna` before I got into this project. It's pretty cool!